include/deal.II/base/symmetric_tensor.h

```diff
@@ -3544,6 +3544,121 @@ DEAL_II_HOST constexpr inline DEAL_II_ALWAYS_INLINE
   return result;
 }
 
+/**
+ * An enumeration to describe the desired symmetry of the
+ * symmetric fourth order tensor
+ */
+enum class SymmetryType
+{
+  /**
+   * Only minor symmetry of the form:
+   * $A_{ijkl}=A_{jikl}=A_{ijlk}=A_{jilk}$.
+   */
+  minor,
+  /** 
+   * Both minor and major symmetry of the form:
+   * Minor - $A_{ijkl}=A_{jikl}=A_{ijlk}=A_{jilk}$
+   * Major - $A_{ijkl}=A_{klij}$.
+   */
+  minor_major,
+};
+
+/**
+ * Return the symmetrized version of a full rank-4 tensor, i.e.
+ * as a symmetric rank-4 tensor.
+ * @param symmetry_type decides the type of symmetry.
```

Comment:
Doxygen shows the name of the argument separate from the text that follows. So perhaps this?
```suggestion
 * @param symmetry_type This argument decides the type of symmetry.
```